src/harness/testcases/WINNF_FT_S_FDB_testcase.py

```diff
@@ -0,0 +1,1240 @@
+      grant_ids.append(resp['grantId'])
+
+    # Step 6: Send relinquishment request for grant 'G1'
+    relq_requests = {'relinquishmentRequest': []}
+    for cbsd_id, grant_id in zip(cbsd_ids, grant_ids):
+      # Update Relinquishment Request content
+      relq_requests['relinquishmentRequest'].append({
+        'cbsdId': cbsd_id,
+        'grantId': grant_id
+      })
+    relq_responses = self._sas.Relinquishment(relq_requests)['relinquishmentResponse']
+    self.assertEqual(len(relq_responses), len(relq_requests['relinquishmentRequest']))
+
+    # Check relinquishment response
+    for relq_resp in relq_responses:
+      self.assertEqual(relq_resp['response']['responseCode'], 0)
+
+    del relq_requests, relq_responses
+    del grant_request_g1, grant_response_g1
+
+    # TOD0
+    # Step 7: Modify the GWBL database to include a modified version of the GWBL 'W'.
+    # 'W' is moved further than 150 kms from FSS site.
+
+    # Step 8: Trigger daily activities
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Step 9: Send Grant Request 'G2'
+    # Forming of 'G2' grants for all the configured grants
+    addCbsdIdsToRequests(cbsd_ids,config['grantRequests'][1])
+    grant_request_g2 = {'grantRequest': config['grantRequests'][1]}
+
+    grant_response_g2 = self._sas.Grant(grant_request_g2)['grantResponse']
+
+    # Check the grant response code is as expected (SUCCESS or INTERFERENCE).
+    for resp in grant_response_g2:
+      self.assertIn(resp['response']['responseCode'], config['expectedResponseCodes'][1])
+
+    del grant_request_g2, grant_response_g2
+
+  def generate_FDB_7_default_config(self, filename):
+    """Generates the WinnForum configuration for FDB.7"""
+
+    # Load devices info
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+
+    # Update the FCC ID of CBSDs with the FCC ID 'F_ID' from FCC ID database.
+    device_a['fccId'] = "F_ID"
+    device_b['fccId'] = "F_ID"
+
+    # Creating conditionals for Cat B devices
+    self.assertEqual(device_b['cbsdCategory'], 'B')
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['installationParam']
+    del device_b['measCapability']
+
+    conditionals = [conditionals_b]
+
+    # Create the actual config.
+    config = {
+        'registrationRequests': [device_a, device_b],
+        'expectedResponseCodes': [(103,), (0,)],
+        'conditionalRegistrationData': conditionals
```

Comment:
To keep things simple, maybe we want to use only Cat A devices in this test case. Otherwise we may have trouble injecting the conditional reg data (we can only do it once the FCC ID has been whitelisted by the SAS, but we don't know exactly when that will be). The logic for Cat A vs. Cat B FCC ID validation should be the same so I don't think it poses a problem for the test case integrity. What do you think? Is there an easier solution?

Reply:
FDB.7     Loading CAT A devices only.